code/game/machinery/doors/airlock_electronics.dm

```diff
@@ -53,9 +53,10 @@
 	return data
 
 /obj/item/electronics/airlock/ui_act(action, params)
-	. = ..()
-	if(.)
-		return
+	if(!params["rcd"]) //if this wasen't call by the rcd then do these checks else skip them
```

Comment:
i do not understand this. why is this here?

Reply:
An optimization when the RCD ui_act calls this method. RCD
```
code/game/objects/items/RCD.dm
775    else
776		params["rcd"] = TRUE
777		airlock_electronics.ui_act(action, params)
```
 There is no need for `airlock_electronics` `ui_act()` to call its super method and waste processing time as it's of no use if the RCD calls this method. We can just skip these steps

Reply:
This isn't worth it, the cost of a proc call in a function this cold is way too cheap to have this burden. Plus it means a hacker can send the "rcd" parameter and never have to worry about permission checks

Reply:
U right I'll remove it

Reply:
There is a problem tho.  Like u said the permission checks will do a early return if I manually call it from rcd ui_act(). 
I can't move it inside a separate proc either cause again a hacker can just call that proc

Reply:
I don't want to manually copy all the stuff inside ui act and paste it here either cause that would be a big mess

Reply:
I have no solution for this.  Could you suggest something?

Reply:
I'm just gonna move it inside its own proc.  Don't see any other way

Reply:
Now that I think of it don't know why a hacker would call this proc manually when he can just use the interface.  I made a dumb argument nevermind